File: boj/b_1759.py
import sys
sys.stdin = open("input.txt", "r") # 제거
input = sys.stdin.readline

l, c = map(int, input().split())
arr = sorted(list(input().split()))
def dfs(s):
    if len(s) == l:
        a, b = 0, 0
        for m in s:
            if m in "aeiou":
                a += 1
            else:
                b += 1
        if a >= 1 and b >= 2:
            return print("".join(s))
        return
    for v in arr:
        if v > s[-1]:
            s.append(v)
            dfs(s)
            s.pop()
for i in range(c - l + 1):
    dfs([arr[i]])

File: boj/b_1759.js
function solution(l, c, arr) {
  const dfs = (s) => {
    if (s.length === l) {
      let [a, b] = [0, 0];
      for (const m of s) "aeiou".includes(m) ? (a += 1) : (b += 1);
      if (a >= 1 && b >= 2) return console.log(s.join(""));
      return;
    }
    for (const v of arr) {
      if (v > s[s.length - 1]) {
        s.push(v);
        dfs(s);
        s.pop();
      }
    }
  };
  arr.sort();
  for (let i = 0; i < c - l + 1; i += 1) dfs([arr[i]]);
}

solution(4, 6, ["a", "t", "c", "i", "s", "w"]);
